CI dependencies are broken

See test PR #38065 
https://drone.owncloud.com/owncloud/core/27424/1/4
```
behat composer.lock is not up to date.
composer bin behat install --no-progress
No lock file found. Updating dependencies instead of installing from lock file. Use composer update over composer install if you do not have a lock file.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - sensiolabs/behat-page-object-extension[v2.3.0, ..., v2.3.2] require ocramius/proxy-manager ^2.1.1 -> satisfiable by ocramius/proxy-manager[2.1.1, ..., 2.9.1].
    - ocramius/proxy-manager[2.2.0, ..., 2.2.3] require ocramius/package-versions ^1.1.3 -> satisfiable by ocramius/package-versions[1.1.3, ..., 1.11.0].
    - ocramius/proxy-manager 2.1.1 requires ocramius/package-versions ^1.1.1 -> satisfiable by ocramius/package-versions[1.1.1, ..., 1.11.0].
    - ocramius/package-versions[1.1.1, ..., 1.2.0] require composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - ocramius/package-versions[1.3.0, ..., 1.4.2] require composer-plugin-api ^1.0.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - ocramius/package-versions[1.5.0, ..., 1.5.1] require php ^7.3.0 -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/package-versions[1.6.0, ..., 1.9.0] require php ^7.4.0 -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/proxy-manager[2.3.0, ..., 2.5.2] require php ^7.4.0 -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/package-versions[1.10.0, ..., 1.11.0] require php ^7.4.7 -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/proxy-manager[2.6.0, ..., 2.7.0] require php 7.4.* -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/proxy-manager 2.7.1 requires php 7.4.0 -> your php version (7.2.34) does not satisfy that requirement.
    - ocramius/proxy-manager[2.8.0, ..., 2.9.1] require php ~7.4.1 -> your php version (7.2.34) does not satisfy that requirement.
    - Root composer.json requires sensiolabs/behat-page-object-extension ^2.3 -> satisfiable by sensiolabs/behat-page-object-extension[v2.3.0, v2.3.1, v2.3.2].

ocramius/package-versions only provides support for Composer 2 in 1.8+, which requires PHP 7.4.
If you can not upgrade PHP you can require composer/package-versions-deprecated to resolve this with PHP 7.0+.

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
make: *** [vendor-bin/behat/vendor] Error 2
```

--- a/.drone.star
+++ b/.drone.star
@@ -2026,7 +2026,21 @@ def vendorbinInstall(phpVersion):
 			'COMPOSER_HOME': '/drone/src/.cache/composer'
 		},
 		'commands': [
-			'make vendor-bin-deps'
+			'make vendor-bin-codestyle',
+			'make vendor-bin-codesniffer',
+			'make vendor-bin-phan',
+			'make vendor-bin-phpstan'
+		]
+	},
+	{
+		'name': 'vendorbin-behat',
+		'image': 'owncloudci/php:7.4',
+		'pull': 'always',
+		'environment': {
+			'COMPOSER_HOME': '/drone/src/.cache/composer'
+		},
+		'commands': [
+			'make vendor-bin-behat'
 		]
 	}]
 

--- a/.drone.star
+++ b/.drone.star
@@ -375,7 +375,11 @@ def dependencies(ctx):
 				'steps':
 					cacheRestore() +
 					composerInstall(phpVersion) +
-					vendorbinInstall(phpVersion) +
+					vendorbinCodestyle(phpVersion) +
+					vendorbinCodesniffer(phpVersion) +
+					vendorbinPhan(phpVersion) +
+					vendorbinPhpstan(phpVersion) +
+					vendorbinBehat() +
 					yarnInstall(phpVersion) +
 					cacheRebuildOnEventPush() +
 					cacheFlushOnEventPush(),
@@ -442,7 +446,8 @@ def codestyle():
 				'steps':
 					cacheRestore() +
 					composerInstall(phpVersion) +
-					vendorbinInstall(phpVersion) +
+					vendorbinCodestyle(phpVersion) +
+					vendorbinCodesniffer(phpVersion) +
 					yarnInstall(phpVersion) +
 					[
 						{
@@ -618,7 +623,7 @@ def phpstan():
 				'steps':
 					cacheRestore() +
 					composerInstall(phpVersion) +
-					vendorbinInstall(phpVersion) +
+					vendorbinPhpstan(phpVersion) +
 					yarnInstall(phpVersion) +
 					installServer(phpVersion, 'sqlite', params['logLevel']) +
 				[
@@ -692,7 +697,7 @@ def phan():
 				'steps':
 					cacheRestore() +
 					composerInstall(phpVersion) +
-					vendorbinInstall(phpVersion) +
+					vendorbinPhan(phpVersion) +
 					yarnInstall(phpVersion) +
 					installServer(phpVersion, 'sqlite', params['logLevel']) +
 				[
@@ -775,7 +780,6 @@ def litmus():
 				'steps':
 					cacheRestore() +
 					composerInstall(phpVersion) +
-					vendorbinInstall(phpVersion) +
 					yarnInstall(phpVersion) +
 					installServer(phpVersion, db, params['logLevel'], params['useHttps']) +
 					setupLocalStorage(phpVersion) +
@@ -943,7 +947,6 @@ def dav():
 					'steps':
 						cacheRestore() +
 						composerInstall(phpVersion) +
-						vendorbinInstall(phpVersion) +
 						yarnInstall(phpVersion) +
 						installServer(phpVersion, db, params['logLevel']) +
 						davInstall(phpVersion, scriptPath) +
@@ -1015,7 +1018,6 @@ def javascript(ctx):
 		'steps':
 			cacheRestore() +
 			composerInstall(params['phpVersion']) +
-			vendorbinInstall(params['phpVersion']) +
 			yarnInstall(params['phpVersion']) +
 		[
 			{
@@ -1190,7 +1192,6 @@ def phptests(ctx, testType):
 						'steps':
 							cacheRestore() +
 							composerInstall(phpVersion) +
-							vendorbinInstall(phpVersion) +
 							yarnInstall(phpVersion) +
 							installServer(phpVersion, db, params['logLevel']) +
 							installExtraApps(phpVersion, extraAppsDict) +
@@ -1475,7 +1476,7 @@ def acceptance(ctx):
 									'steps':
 										cacheRestore() +
 										composerInstall(phpVersion) +
-										vendorbinInstall(phpVersion) +
+										vendorbinBehat() +
 										yarnInstall(phpVersion) +
 										installServer(phpVersion, db, params['logLevel'], params['useHttps'], params['federatedServerNeeded'], params['proxyNeeded']) +
 										(
@@ -1544,7 +1545,6 @@ def sonarAnalysis(ctx, phpVersion = '7.4'):
 		'steps':
 			cacheRestore() +
 			composerInstall(phpVersion) +
-			vendorbinInstall(phpVersion) +
 			yarnInstall(phpVersion) +
 			installServer(phpVersion, 'sqlite') +
 		[
@@ -2017,22 +2017,60 @@ def composerInstall(phpVersion):
 		]
 	}]
 
-def vendorbinInstall(phpVersion):
+def vendorbinCodestyle(phpVersion):
 	return [{
-		'name': 'vendorbin-install',
+		'name': 'vendorbin-codestyle',
 		'image': 'owncloudci/php:%s' % phpVersion,
 		'pull': 'always',
 		'environment': {
 			'COMPOSER_HOME': '/drone/src/.cache/composer'
 		},
 		'commands': [
-			'make vendor-bin-codestyle',
-			'make vendor-bin-codesniffer',
-			'make vendor-bin-phan',
+			'make vendor-bin-codestyle'
+		]
+	}]
+
+def vendorbinCodesniffer(phpVersion):
+	return [{
+		'name': 'vendorbin-codesniffer',
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'environment': {
+			'COMPOSER_HOME': '/drone/src/.cache/composer'
+		},
+		'commands': [
+			'make vendor-bin-codesniffer'
+		]
+	}]
+
+def vendorbinPhan(phpVersion):
+	return [{
+		'name': 'vendorbin-phan',
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'environment': {
+			'COMPOSER_HOME': '/drone/src/.cache/composer'
+		},
+		'commands': [
+			'make vendor-bin-phan'
+		]
+	}]
+
+def vendorbinPhpstan(phpVersion):
+	return [{
+		'name': 'vendorbin-phpstan',
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'environment': {
+			'COMPOSER_HOME': '/drone/src/.cache/composer'
+		},
+		'commands': [
 			'make vendor-bin-phpstan'
 		]
-	},
-	{
+	}]
+
+def vendorbinBehat():
+	return [{
 		'name': 'vendorbin-behat',
 		'image': 'owncloudci/php:7.4',
 		'pull': 'always',

--- a/Makefile
+++ b/Makefile
@@ -387,6 +387,21 @@ vendor/bamarni/composer-bin-plugin: composer.lock
 .PHONY: vendor-bin-deps
 vendor-bin-deps: vendor-bin/owncloud-codestyle/vendor vendor-bin/php_codesniffer/vendor vendor-bin/phan/vendor vendor-bin/phpstan/vendor vendor-bin/behat/vendor
 
+.PHONY: vendor-bin-codestyle
+vendor-bin-codestyle: vendor-bin/owncloud-codestyle/vendor
+
+.PHONY: vendor-bin-codesniffer
+vendor-bin-codesniffer: vendor-bin/php_codesniffer/vendor
+
+.PHONY: vendor-bin-phan
+vendor-bin-phan: vendor-bin/phan/vendor
+
+.PHONY: vendor-bin-phpstan
+vendor-bin-phpstan: vendor-bin/phpstan/vendor
+
+.PHONY: vendor-bin-behat
+vendor-bin-behat: vendor-bin/behat/vendor
+
 vendor-bin/owncloud-codestyle/vendor: vendor/bamarni/composer-bin-plugin vendor-bin/owncloud-codestyle/composer.lock
 	composer bin owncloud-codestyle install --no-progress
 

